### PR TITLE
Refactor help-center components and add chat history feature

### PR DESCRIPTION
**Pull request by agrullon95**
| Recent Conversations  | Support History |
| ------------- | ------------- |
| Screenshot 2024-10-18 at 16 08 39 | Screenshot 2024-10-18 at 16 08 32 |

Fixes https://github.com/Automattic/dotcom-forge/issues/9275
Display recent conversations and support history.
This work is behind feature flag `help-center-experience`

## Proposed Changes

* Check for unread conversations and messages. 
* Display recent conversations that lead to support history route
* Display conversations in support history and open conversation on click

## Why are these changes being made?

*

## Testing Instructions
- Checkout this PR and build it locally
- Add the ff in the url `?flags=help-center-experience`
- Create a conversation and type a message
- Locate the relevant ticket in the support sandbox
- Assign to yourself and reply. This will create an unread message in the conversation
- Checkout how that is rendered
- Test various conditions here, with the read/unread count
- Clicking on a recent conversation should navigate to support-history where all chats are displayed
- Clicking on a chat there should open the conversation and allow you to reply

*

### CODE REVIEW

File: packages/help-center/src/components/help-center.tsx
Comment (heavyweight) [inline packages/help-center/src/components/help-center.tsx:1]: This file should be reverted before merge.
Was added to wire smooch.
It doesn't use feature flags.

File: packages/zendesk-client/src/constants.ts
Comment (heavyweight) [inline packages/zendesk-client/src/constants.ts:1]: This file should be reverted before merge.
Added to wire smooch

File: packages/zendesk-client/src/use-smooch.ts
Comment (heavyweight) [inline packages/zendesk-client/src/use-smooch.ts:1]: This file should be reverted before merge.
Added to wire smooch.

**Comment by escapemanuele** [comment]
Just a reminder to use the `help-center-experience` flag

File: packages/odie-client/src/data/use-get-zendesk-conversation.ts [hunk 1]
```diff
@@ -1,28 +1,47 @@
...
-		return {
-			content: message.text,
-			role: message.role,
-			type: message.type === 'text' ? 'message' : message.type,
-		};
-	} );
-
-	return { ...conversation, messages };
+	// We need to ensure that more than one message is loaded
+	return Smooch.getConversationById( conversation.id )
```
Comment (escapemanuele) [inline packages/odie-client/src/data/use-get-zendesk-conversation.ts:41]: Is this `Smooch.getConversationById( conversation.id )` different than the `conversation` we retrieved before?

File: packages/help-center/src/components/help-center-recent-conversations.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+import { HELP_CENTER_STORE } from '../stores';
+import { HelpCenterSupportChatMessage } from './help-center-support-chat-message';
+import type { ZendeskConversation } from '@automattic/odie-client';
+
+import './help-center-recent-conversations.scss';
+
+const GetSectionName = ( unreadCount: number ) => {
+	const { __ } = useI18n();
+	if ( unreadCount > 1 ) {
+		return __( 'Unread Conversation', __i18n_text_domain__ );
```
Comment (escapemanuele) [inline packages/help-center/src/components/help-center-recent-conversations.tsx]: Should this be 'Unread Conversations' ?

File: packages/odie-client/src/data/use-get-zendesk-conversation.ts (hunk 1 above)
Comment (heavyweight) [reply packages/odie-client/src/data/use-get-zendesk-conversation.ts:41]: It depends.
On first load all conversations will have only the last message in them.
This is to ensure that we retrieve the messages, and don't just display the last one.

**Review by escapemanuele — APPROVED** [review]
What a beautiful work! 👏

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16940471

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @agrullon95 for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16940471).

**Pull request merged**